device-telemetry/test/services/test/storage/timeSeries/TimeSeriesClientTest.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+package services.test.storage.timeSeries;
+
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.Status;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.exceptions.InvalidConfigurationException;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.runtime.*;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.storage.timeSeries.ITimeSeriesClient;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.storage.timeSeries.TimeSeriesClient;
+import helpers.UnitTest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import play.libs.ws.WSClient;
+
+import static org.mockito.Mockito.mock;
+
+public class TimeSeriesClientTest {
+
+    private WSClient wsClient;
+    private IServicesConfig servicesConfig;
+    private ITimeSeriesClient timeSeriesClient;
+
+    @Before
+    public void setUp() throws InvalidConfigurationException {
+        this.wsClient = mock(WSClient.class);
+        this.servicesConfig = new ServicesConfig(
+            "http://localhost:9022/v1",
+            mock(MessagesConfig.class),
+            mock(AlarmsConfig.class)
+        );
+        this.timeSeriesClient = new TimeSeriesClient(this.servicesConfig, this.wsClient);
+    }
+
+    @Test(timeout = 50000)
+    @Category({UnitTest.class})
+    public void PingReturnsFalse_WhenConfigValuesAreNull() throws Throwable {
+        // Act & Assert
+        Status result = this.timeSeriesClient.ping();
+
+        // Assert
+        Assert.assertFalse(result.isHealthy());
+    }
+
+    @Test(timeout = 50000, expected = InvalidConfigurationException.class)
+    @Category({UnitTest.class})
+    public void QueryThrows_IfInvalidAuthParams() throws Throwable {
```

Comment:
Do you need more tests for queryEvents? Like on success?

Reply:
WSClient and AAD token is not easily to set up for testing purpose for now.  This can be done by integration tests, I think.